### ISSUE TITLE
LayerNorm produces incorrect output with elementwise_affine=False

### ISSUE THREAD

**Issue by raghukiran1224** [torch-spyre/torch-spyre#1980]
## Summary

`nn.LayerNorm` with `elementwise_affine=False` produces incorrect (constant) output on Spyre. The same operation with `elementwise_affine=True` (default) works correctly.

## Minimal Reproduction

```python
import torch
import torch.nn as nn

torch.manual_seed(42)
x = torch.randn(1, 4096, dtype=torch.float16)

# This works correctly
ln_affine = nn.LayerNorm(4096).half().to("spyre")
compiled_affine = torch.compile(ln_affine)
result_affine = compiled_affine(x.to("spyre")).cpu()
# result_affine matches CPU output

# This produces garbage output
ln_no_affine = nn.LayerNorm(4096, elementwise_affine=False).half().to("spyre")
compiled_no_affine = torch.compile(ln_no_affine)
result_no_affine = compiled_no_affine(x.to("spyre")).cpu()
# result_no_affine is a constant value (-0.0013) for every element
```

## Observed Behavior

With `elementwise_affine=False`, every output element is approximately `-0.0013`:

```
Spyre fp16:  [-0.0013, -0.0013, -0.0013, -0.0013, -0.0013, -0.0013, -0.0013, -0.0013]
CPU fp16:    [0.3899, 1.4580, 0.1533, -1.0928, 0.6699, -0.1923, 1.8340, -0.4834]
```

- Max abs diff vs CPU fp16: **3.57**
- 100% of elements differ from CPU

## Expected Behavior

Output should match CPU fp16 within normal fp16 precision bounds (max diff ~0.008 as seen with `elementwise_affine=True`).

## Environment

- SENCORES=1
- dtype: float16
- batch size: 1
- normalized_shape: [4096]

**Comment by raghukiran1224** [comment]
## Update: Root cause identified and fix verified

**Root cause:** The DDL template `layernormnorm.ddl` always expects 5 inputs (x, mean, norm_mean, weight, bias) and performs `output = normalized * weight + bias`. When `elementwise_affine=False`, the lowering passes only 3 inputs (dropping weight/bias), so the backend reads uninitialized memory.

**Fix:** In the decomposition, synthesize identity tensors when weight/bias are None:
```python
if weight is None:
    weight = torch.ones(normalized_shape, dtype=input.dtype, device=input.device)
if bias is None:
    bias = torch.zeros(normalized_shape, dtype=input.dtype, device=input.device)
```

Verified on hardware — max diff drops from 3.57 to 0.0078 (matching the elementwise_affine=True precision).

**Priority note:** This is low priority. Real models (GPT, LLaMA, etc.) all use `elementwise_affine=True` (the default), which works correctly. The `elementwise_affine=False` variant is rare (some diffusion models, research experiments).

**Comment by raghukiran1224** [comment]
Re-triage: **cannot reproduce — resolved.** Verified on pod rganti-spyre-main at torch-spyre `9be4e3d` (current main).

`F.layer_norm(x, x.shape[1:], weight=None, bias=None)` (the `elementwise_affine=False` condition) now compiles and matches the CPU reference on (8,4096) fp16. Note: the suite `test_layernorm_cpu` only exercises weight!=None/bias=0, so this was confirmed with a dedicated affine=False repro. Closing as fixed.
